completion/Cargo.toml

```diff
@@ -0,0 +1,21 @@
+[package]
+name = "gluon_completion"
+version = "0.5.0" # GLUON
+authors = ["Markus <marwes91@gmail.com>"]
+
+license = "MIT"
+
+description = "Auto-completion for the gluon programming language"
+
+repository = "https://github.com/gluon-lang/gluon"
+documentation = "https://docs.rs/gluon"
+
+[dependencies]
+gluon_base = { path = "../base", version = "0.5.0" } # GLUON
+
+[dev-dependencies]
+collect-mac = "0.1.0"
+env_logger = { version = "0.3.4" }
+
+gluon_check = { path = "../check", version = "0.5.0" } # GLUON
+gluon_parser = { path = "../parser", version = "0.5.0" } # GLUON
```

Comment:
In other crates, you use optional dependencies sometimes for things that are only used in tests, and activate them using features. I couldn't find good info on doing that vs just making them dev-dependencies. Should I switch these to be optional and activated for a feature named `test`?

Reply:
They should be dev-dependencies where it is possible. However, at least in the case of the `gluon_parser` crate it was necesssary to provide [this](https://github.com/gluon-lang/gluon/blob/e5671cabae13a44ee491994845bb536c317f9a34/parser/src/lib.rs#L416-L422) function specialization (where `Ident=String`). That function was only required for testing but to use it in tests defined in the `tests` directory I couldn't use plain `cfg(test)` since items `cfg` by that do not appear in the exported (`gluon_parser`) library. (Why I needed the specialization was because `gluon_parser` used to take 10+ minutes to compile in LLVM so I really needed it to be specialized immediately to avoid that cost when compiling gluon_check, etc).

Sorry if that is a bit confusing. It is fine to leave them as optional if like since --features=test will be necessary regardless for the time being at least.

Reply:
If you have already moved them to `dev-dependencies` and all the tests pass (which they do on travis!) then that is better than optional dependencies so leave them as is. There is simply no real need to go over and move other optional dependencies to dev since there is at least a couple of them which can't be a `dev-dependency`. So thus we are still forced to pass `--features=test` to run all the tests. :)